### ISSUE TITLE
G23 — Adapter Sweep: Stanford Cars

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#31]
## Gauntlet 23 — Adapter Sweep: Stanford Cars

**Phase:** Phase 4 – Exploration
**Kill Condition:** In-domain top-1 accuracy on Stanford Cars (ViT-L/14, 1024d) with Cars-trained adapter is lower than Dogs adapter baseline on Dogs (G11 reference). Adapter must demonstrate domain-specific metric improvement.

**G11 Reference:** Full stack (adapter+NSTP+v=0.80) achieved 99.71% accuracy, condensation ratio 0.47 on Stanford Dogs subset (ImageNet classes 151–200).

---

### Objective

Create a self-contained training + evaluation script `benchmarks/g23_adapter_cars.py` that:

1. **Downloads Stanford Cars** dataset — 196 car classes, ~8.1k train / ~8.0k test images. Use `torchvision.datasets.StanfordCars` (available in torchvision ≥ 0.13) or manual download + ImageFolder.
2. **Extracts DINOv2 ViT-L/14 embeddings** (1024d) for all train and test images. Cache to `.pt` files (`stanford_cars_train.pt`, `stanford_cars_test.pt`) with `{"embeds": Tensor, "labels": Tensor}` format.
3. **Trains a domain-specific MetricAdapter** on Stanford Cars train embeddings using the same architecture and protocol as `train_adapter_triplet.py`:
   - `MetricAdapter(input_dim=1024, output_dim=256, hidden_dim=512)`
   - Online Hard Negative Mining (OHNM) triplet loss
   - 5 epochs, batch_size=128, lr=1e-4, margin=0.2
   - Save weights to `adapter_stanford_cars.pt`
4. **Evaluates the full stack** (adapter + NSTP + vigilance sweep) on Stanford Cars test set:
   - Vigilance sweep: `[0.92, 0.85, 0.82, 0.80]`
   - For each vigilance level, report: top-1 accuracy, prototype count, condensation ratio
   - Also evaluate baseline (no adapter, v=0.92) for comparison
5. **Kill condition check:** Compare best Cars-adapter accuracy against G11 Dogs-adapter reference (99.71%). If Cars accuracy < Dogs reference, kill condition met.

### Implementation Details

- Import from existing modules: `adapter.MetricAdapter`, `fast_associative_memory.FastAssociativeMemory`, `nstp.NSTPController`
- Follow the same DINOv2 extraction pattern as `extract_imagenet_r_vitl14.py` (ViT-L/14, 1024d)
- Follow the same training loop structure as `train_adapter_triplet.py` (OHNM miner + triplet loss)
- Follow the same evaluation pattern as `benchmarks/benchmark_mechanisms.py` for the FAM ingest + eval loop
- Condensation ratio = num_prototypes / num_train_samples
- If G22 (`benchmarks/g22_adapter_birds.py`) has already been merged, feel free to reuse shared helper functions from it. If not, implement standalone.

### CLI Interface

```
python benchmarks/g23_adapter_cars.py [--root DATA_DIR] [--epochs 5] [--batch-size 128] [--lr 1e-4] [--margin 0.2] [--device auto]
```

- `--root`: Data directory for Stanford Cars download/cache (default: `./data/stanford_cars`)
- All other args match `train_adapter_triplet.py` conventions

### Expected Output

```
G23 — Adapter Sweep: Stanford Cars
===================================================================
  Config                          Accuracy    Protos   Cond.Ratio
  -------------------------------------------------------------------
  baseline (no adapter, v=0.92)   XX.XX%      NNNN     X.XX
  cars adapter (v=0.92)           XX.XX%      NNNN     X.XX
  cars adapter (v=0.85)           XX.XX%      NNNN     X.XX
  cars adapter (v=0.82)           XX.XX%      NNNN     X.XX
  cars adapter (v=0.80)           XX.XX%      NNNN     X.XX
===================================================================

Kill condition: Cars best accuracy XX.XX% vs Dogs reference 99.71%
→ PASS / KILL
```

### Files to Create
- `benchmarks/g23_adapter_cars.py` — main script (train + eval + report)

### Files to Reference (do not modify)
- `adapter.py` — `MetricAdapter` class
- `train_adapter_triplet.py` — OHNM training protocol reference
- `fast_associative_memory.py` — `FastAssociativeMemory` class
- `nstp.py` — `NSTPController` class
- `benchmarks/benchmark_mechanisms.py` — evaluation loop reference
